src/test/ui/const-generics/different_byref.stderr

```diff
@@ -0,0 +1,20 @@
+warning: the feature `const_generics` is incomplete and may cause the compiler to crash
+  --> $DIR/different_byref.rs:1:12
+   |
+LL | #![feature(const_generics)]
+   |            ^^^^^^^^^^^^^^
+   |
+   = note: `#[warn(incomplete_features)]` on by default
+
+error[E0308]: mismatched types
+  --> $DIR/different_byref.rs:8:9
+   |
+LL |     x = Const::<{ [4] }> {};
+   |         ^^^^^^^^^^^^^^^^^^^ expected `3usize`, found `4usize`
+   |
+   = note: expected struct `Const<ByRef { alloc: Allocation { bytes: [3, 0, 0, 0, 0, 0, 0, 0], relocations: Relocations(SortedMap { data: [] }), undef_mask: UndefMask { blocks: [255], len: Size { raw: 8 } }, size: Size { raw: 8 }, align: Align { pow2: 3 }, mutability: Not, extra: () }, offset: Size { raw: 0 } }: [usize; 1]>`
+              found struct `Const<ByRef { alloc: Allocation { bytes: [4, 0, 0, 0, 0, 0, 0, 0], relocations: Relocations(SortedMap { data: [] }), undef_mask: UndefMask { blocks: [255], len: Size { raw: 8 } }, size: Size { raw: 8 }, align: Align { pow2: 3 }, mutability: Not, extra: () }, offset: Size { raw: 0 } }: [usize; 1]>`
```

Comment:
I have a WIP branch where I make this print the actual values, I think we should wait for that, so that we can actually see it in action.

Reply:
:man_shrugging: don't think that waiting is worth it if this is otherwise ready.

Reply:
It's not ready, because we can't read the error message.

Reply:
We already emit this error message rn. This PR only prevents correct programs from failing with the given error message.

Reply:
This shouldn't be `ByRef`. The layout is `Scalar` for single element arrays, and all `Scalar` constants become `ConstValue::Scalar` in https://github.com/rust-lang/rust/blob/ba72b15666b2491415aec703a02c2364fe5e2790/src/librustc_mir/const_eval/eval_queries.rs#L109

Does this `ty::Const` not come from const eval?

Reply:
> The layout is `Scalar` for single element arrays

Is it though? I don't remember this.

Reply:
Ah no, we just do this for tuples and structs. Slightly inconsistent, but :shrug:

Reply:
That branch I mentioned is now a PR: #71232.